internal/meshdetect/detector.go

```diff
@@ -213,25 +221,94 @@ func isPodInitContainerInjected(meshKind MeshKind, pod *corev1.Pod) bool {
 
 // DetectServiceDistribution detects how many services are running under each mesh.
 func (d *Detector) DetectServiceDistribution(ctx context.Context) (*ServiceDistributionResults, error) {
+
 	ret := &ServiceDistributionResults{
 		MeshDistribution: map[MeshKind]int{},
 	}
 
 	// list all services
-	serviceList := &corev1.ServiceList{}
-	err := d.Client.List(ctx, serviceList)
-	if err != nil {
-		d.Logger.Error(err, "failed to list all services in cluster")
-		return nil, err
+	serviceList := []*corev1.Service{}
+
+	continueListing := true
+	continueToken := ""
+	for continueListing {
+		partialServiceList := &corev1.ServiceList{}
+		err := d.Client.List(ctx, partialServiceList, client.Limit(defaultPageSize), client.Continue(continueToken))
+		if err != nil {
+			d.Logger.Error(err, "failed to list services in cluster")
+			return nil, err
+		}
+
+		for i := range partialServiceList.Items {
+			serviceList = append(serviceList, &partialServiceList.Items[i])
+		}
+
+		continueToken = partialServiceList.GetContinue()
+
+		if partialServiceList.RemainingItemCount == nil || *partialServiceList.RemainingItemCount == 0 {
+			continueListing = false
```

Comment:
@randmonkey Can you please take a look at (and fix) those inneffectual assignments pointed out by the linter?

Reply:
fixed.